src/explorer/glueCatalog.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as vscode from "vscode";
+import { DefaultGlueClient } from "../clients/glueClient";
+import { Database, Table } from "@aws-sdk/client-glue";
+
+export class GlueCatalogNode
+  implements vscode.TreeDataProvider<vscode.TreeItem>
+{
+  private _onDidChangeTreeData: vscode.EventEmitter<
+  GlueCatalogDatabaseNode | undefined | null | void
+  > = new vscode.EventEmitter<
+  GlueCatalogDatabaseNode | undefined | null | void
+  >();
+  readonly onDidChangeTreeData: vscode.Event<
+  GlueCatalogDatabaseNode | undefined | null | void
+  > = this._onDidChangeTreeData.event;
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  public constructor(private readonly glue: DefaultGlueClient) {}
+
+  getTreeItem(element: GlueCatalogDatabaseNode): vscode.TreeItem {
+    return element;
+  }
+
+  async getChildren(
+    element?: GlueCatalogDatabaseNode
+  ): Promise<vscode.TreeItem[]> {
+    if (element) {
+      return Promise.resolve(element.getChildren());
+    } else {
+      const glueDatabases = await this.glue.listDatabases();
+      return Promise.resolve(
+        glueDatabases.map(
+          (glueDatabase: Database) =>
+            new GlueCatalogDatabaseNode(glueDatabase.Name!, this.glue)
+        )
+      );
+    }
+  }
+}
+
+export class GlueCatalogDatabaseNode extends vscode.TreeItem {
+  constructor(
+    private readonly databaseName: string,
+    private readonly glue: DefaultGlueClient
+  ) {
+    super(databaseName, vscode.TreeItemCollapsibleState.Collapsed);
+    this.contextValue = "Glue Database";
+  }
+
+  getTreeItem(element: GlueCatalogTable): vscode.TreeItem {
+    return element;
+  }
+
+  getChildren(
+    element?: GlueCatalogTable
+  ): Thenable<GlueCatalogTable[]> {
+    return Promise.resolve(
+      this.glue
+        .listTables(this.databaseName)
+        .then((tables) =>
+          tables.map((table) => new GlueCatalogTable(table))
+        )
+    );
+  }
+
+}
+
+export class GlueCatalogTable extends vscode.TreeItem {
+  constructor(
+    private readonly gluetable: Table
+  ) {
+    super(`${gluetable.Name} v:${gluetable.VersionId}`);
```

Comment:
I find the `v:` a little distracting. Maybe we can tweak it a bit so it's shown as the description on the table and remove the `:`.

```suggestion
    super(`${gluetable.Name}`);
    this.description = `v${gluetable.VersionId}`;
```

Reply:
Ok, we can in future release a version explorer in the web view. For now we always display the latest.